wiki/Gameplay/Full_combo/en.md

```diff
@@ -6,10 +6,21 @@ tags:
 
 # Full combo
 
-**Full combo** (or *FC* for short) is a term used to describe a player obtaining the maximum possible [combo](/wiki/Beatmapping/Combo) on a [beatmap](/wiki/Beatmap). Full combos are scored by passing a beatmap with no misses<!-- TODO: link -->, [sliderbreaks](/wiki/Gameplay/Judgement/Slider_break), or dropped [slider ends](/wiki/Gameplay/Hit_object/Slider/Slidertail).
+**Full combo** (or *FC* for short) is a term used to describe a player passing a [beatmap](/wiki/Beatmap) without ever combo breaking. Full combos are scored by passing a beatmap with no misses<!-- TODO: link --> or [sliderbreaks](/wiki/Gameplay/Judgement/Slider_break).
 
 Scores that lost combo only via dropped slider ends are widely considered by the community to be full combos. This differs from the game client and website's display.
 
 ![Screenshot of full combo and broken combo scores](img/combo-comparison.png "The top score is a full combo, and the bottom score is a broken combo.")
 
 Because the [score](/wiki/Gameplay/Score) of an individual object in [osu!](/wiki/Game_mode/osu!) and [osu!catch](/wiki/Game_mode/osu!catch) heavily depends on combo multiplier and is virtually unbounded, full combos typically award the most score in these game modes.
+
+# Perfect full combo
+**Perfect full combo** (or *PFC* for short) is a term used to describe the game acknowledging a full combo ocurred by coloring the combo green. This is crucial for Hush-Hush medals, as whenever they are checking for the full combo conditional directly, it is referring to a perfect full combo. In all 4 game modes, Standard, Taiko, Catch, and Mania, the way this is technically processed varies:
+
+Standard - Pass a beatmap with no misses, sliderbreaks, or dropped slider ends.
+
+Taiko - Pass a beatmap with no misses. The beatmap must not contain sliders/drumrolls or spinners/dendens/shakers. Taiko skill medals however allow sliders/drumrolls or spinners/dendens/shakers.
```

Comment:
Your primary definition for "full combo" is not what most players see. Most players ignore missed slider ends, ignore if a score has a 50 or 100 in mania if it has 0 combo breaks, and most importantly, completely ignore if the game says if a score is a so called "full combo" or not. So long as a score does not have a combo break, it is an FC in most players' eyes. If anything, players say "this player full combo'd but missed n slider ends". The game's way of seeing an FC is completely different from how players see it, with the only exception being Catch where FC and PFC are identical.

Perfect full combo refers to what the game sees. It is a boolean in each score that says whether the player "full combo'd" or not.

I looked at what counts as perfect full combos on the following Taiko maps just to confirm what I said (I was wrong about sliders/drumrolls). Feel free to check the leaderboards on this map to see that not all Taiko maps are perfectly full combo-able:

https://osu.ppy.sh/beatmapsets/422261#taiko/913008 - Purely circles => PFC is possible as expected.

https://osu.ppy.sh/beatmapsets/189825#taiko/453680 - Purely circles **but with (at least) 1 slider => PFC is possible, so what I typed regarding it is incorrect.**

https://osu.ppy.sh/beatmapsets/422261#taiko/913007 - Purely circles **but with (at least) 1 spinner => PFC is impossible as expected.**

So whoops, both circles and sliders/drumrolls are perfectly fine for a (green) perfect full combo in Taiko. **The issue purely comes from spinners/dendens/shakers.**

That is very interesting. I theorize it is because Taiko counts spinners/dendens/shakers as 1 combo each, but since they give no combo, a PFC is impossible. However, it is very bizarre Taiko accounts for sliders/drumrolls perfectly fine, but not spinners/dendens/shakers. 🤔 

So this is what is what the text should say:
```suggestion
Taiko - Pass a beatmap with no misses. The beatmap must not contain spinners/dendens/shakers. Taiko skill medals however allow spinners/dendens/shakers.
```

Reply:
> https://osu.ppy.sh/beatmapsets/422261#taiko/913007 - Purely circles but with (at least) 1 spinner => PFC is impossible as expected.

this last example looks like a bug and should be reported to osu-web (no green combo count) and osu-stable-issues (no "Perfect" text on the results screen)

> Your primary definition for "full combo" is not what most players see. Most players ignore missed slider ends, ignore if a score has a 50 or 100 in mania if it has 0 combo breaks, and most importantly, completely ignore if the game says if a score is a so called "full combo" or not. So long as a score does not have a combo break, it is an FC in most players' eyes. If anything, players say "this player full combo'd but missed n slider ends". The game's way of seeing an FC is completely different from how players see it, with the only exception being Catch where FC and PFC are identical.

i'm not refuting this - it's perfectly fine to describe the different terms and how and where they're used, but please prioritise game mechanics. even if players use many different terms, the wiki should not be ambiguous about how the game works and what it displays to the player. see [clayton's comment above](https://github.com/ppy/osu-wiki/pull/10285#pullrequestreview-1696011110)

many osu! players come from different backgrounds and play other rhythm games, many of which have different terms and definitions for game mechanics, but osu! (the game) never makes a distinction between "perfect full combo"/"perfect combo" and "full combo". it's entirely a community concept

for reference:

| query | place | results |
| :-- | :-- | --: |
| ["osu" "fc"](https://www.google.com/search?q=%22osu%22+%22fc%22) | google | 7,630,000 |
| ["osu" "pfc"](https://www.google.com/search?q=%22osu%22+%22pfc%22) | google | 1,910,000 |
| ["osu" "full combo"](https://www.google.com/search?q=%22osu%22+%22full+combo%22) | google | 65,800 |
| ["osu" "perfect combo"](https://www.google.com/search?q=%22osu%22+%22perfect+combo%22) | google | 60,400 |
| ["osu" "perfect full combo"](https://www.google.com/search?q=%22osu%22+%22perfect+full+combo%22) | google | 5,910 |
| ["fc"](https://osu.ppy.sh/home/search?mode=forum_post&query=%22fc%22) | osu! forums | 99+ |
| ["full combo"](https://osu.ppy.sh/home/search?mode=forum_post&query=%22full%20combo%22) | osu! forums | 99+ |
| ["perfect combo"](https://www.reddit.com/r/osugame/search/?q=%22perfect%20combo%22&restrict_sr=1&type=comment) | r/osugame comments | 99+ |
| ["perfect combo"](https://osu.ppy.sh/home/search?mode=forum_post&query=%22perfect%20combo%22) | osu! forums | 72 |
| ["pfc"](https://www.reddit.com/r/osugame/search/?q=%22pfc%22&restrict_sr=1&type=comment) | r/osugame comments | 34 |
| ["pfc"](https://osu.ppy.sh/home/search?mode=forum_post&query=%22pfc%22) | osu! forums | 24 |
| ["perfect full combo"](https://osu.ppy.sh/home/search?mode=forum_post&query=%22perfect%20full%20combo%22) | osu! forums | 9 |
| ["perfect full combo"](https://www.reddit.com/r/osugame/search/?q=%22perfect%20full%20combo%22&restrict_sr=1&type=comment) | r/osugame comments | 6 |

Reply:
What do you mean by prioritize game mechanics?

What players normally consider a full combo to be is not a game mechanic as what players call a full combo is not always interpreted as one by the game itself. It is an interpretation players came up with that applies to every other typical rhythm game.

The mechanics of what the game thinks a full combo is is hardcoded into the game already regardless of what players think it is to themselves. (I believe "perfect full combo" came from how in standard, if you hit every whole object without missing any part of it, you perfectly got all the combo, and hence, perfect full combo. In Mania and Taiko it's more confusing.)

For this reason, there should be a way to distinguish between the two of them, one that players acknowledge as a full combo, and one that the game acknowledges is a full combo. It is only natural that players came up with ways to distinguish the two, otherwise, we would always bump into the issue of:

"Yay, I full combo'd this map. But why does the game say it's not an full combo? Well, while you did full combo, that's not what the game considers one to be."
"I full combo'd this map. How come I did not get the full combo medal I was aiming to get? Well, while you did full combo, that's not what the game considers one to be."

---
Taiko seems very fixable, but, it has been out for a long time and so many scores have been set in it. Mania is the same but in a worse state regarding combo. How long have Taiko and Mania have been out? Are you sure that these are bugs that are fixable? I feel like these bugs would have been fixed near the times they released, but these bugs have existed since their very beginnings. You know how many Taiko (and Mania) scores you would need to fix, flipping the full combo boolean to many of them from false to true?

Regardless, if fixing the two modes are doable (with Taiko being easier to fix), especially since soon osu! plans to fully transition to lazer, that would be fantastic.

I have never ever understood why spinners make perfect full combos impossible in Taiko. And I have never understood why 50s or 100s ruin a perfect full combo if you did not combo break in Mania. This is all quite a sophisticated issue, and we all have to somehow adapt to it to prevent any more confusion.